src/v5/encoding.ts

```diff
@@ -160,8 +163,42 @@ export class EncodingHelper {
         ])
     }
 
+    /* L2 Deposit */
+
+    // To comply with: https://github.com/consenlabs/tokenlon-contracts/blob/master/contracts/interfaces/IL2Deposit.sol#L29-L36
+    public encodeL2Deposit(data: L2DepositData): string {
+        const i = new ethers.utils.Interface(abi.L2Deposit)
+        return i.encodeFunctionData("deposit", [
+            [
+                [
+                    data.deposit.l2Identifier,
+                    data.deposit.l1TokenAddr,
+                    data.deposit.l2TokenAddr,
+                    data.deposit.sender,
+                    data.deposit.recipient,
+                    data.deposit.amount,
+                    data.deposit.salt,
+                    data.deposit.expiry,
+                    data.deposit.data,
+                ],
+                data.depositSig,
+            ],
+        ])
+    }
+
     /* Vendor */
 
+    public encodeL2ArbitrumDepositData(user: string, data: L2ArbitrumDepositData): string {
+        return ethers.utils.defaultAbiCoder.encode(
+            ["address", "uint256", "uint256", "uint256"],
```

Comment:
It's better to declare the abi as `const` at the beginning.

Reply:
I tried to add the `const` keyword at the beginning like below, but it shows an error saying that: `A class member cannot have the 'const' keyword.`

```
...
public const encodeL2ArbitrumDepositData(user: string, data: L2ArbitrumDepositData): string {
...
```

Reply:
I found that it can be declared as follows:
```
        const coder = ethers.utils.defaultAbiCoder
        return coder.encode(
        ...
```
I will try to replace it in my next commit, thanks!

Reply:
The abi part is `["uint8", "bytes"]` for example, not the coder.

Reply:
But how I can apply it here with the `const` keyword...
Does it mean that as below..?
```typescript
    public encodeL2ArbitrumDepositData(user: string, data: L2ArbitrumDepositData): string {
        const abi = ["address", "uint256", "uint256", "uint256"]
        return ethers.utils.defaultAbiCoder.encode(abi, [
            user,
            data.maxSubmissionCost,
            data.maxGas,
            data.gasPriceBid,
        ])
    }
```